src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -720,37 +720,30 @@ static int CountDigits(int value)
         // extended attributes. They get collected and saved in that dictionary, with no restrictions.
         private void CollectExtendedAttributesFromStandardFieldsIfNeeded()
         {
-            ExtendedAttributes.Add(PaxEaName, _name);
+            ExtendedAttributes[PaxEaName] = _name;
 
             if (!ExtendedAttributes.ContainsKey(PaxEaMTime))
             {
                 ExtendedAttributes.Add(PaxEaMTime, TarHelpers.GetTimestampStringFromDateTimeOffset(_mTime));
             }
 
-            if (!string.IsNullOrEmpty(_gName))
-            {
-                TryAddStringField(ExtendedAttributes, PaxEaGName, _gName, FieldLengths.GName);
-            }
-
-            if (!string.IsNullOrEmpty(_uName))
-            {
-                TryAddStringField(ExtendedAttributes, PaxEaUName, _uName, FieldLengths.UName);
-            }
+            TryAddStringField(ExtendedAttributes, PaxEaGName, _gName, FieldLengths.GName);
+            TryAddStringField(ExtendedAttributes, PaxEaUName, _uName, FieldLengths.UName);
 
             if (!string.IsNullOrEmpty(_linkName))
```

Comment:
And now that you raised the concern about the mtime case, I think this condition should be changed: We should always synchronize the linkName. The standard field link value will get silently truncated in pax, and the _real_ value will always be located in the dictionary.

The only case we need to treat differently is null: that should be converted to empty string in the dictionary.

Reply:
LinkName does not allow null or empty.

Reply:
https://github.com/dotnet/runtime/blob/43fa22f07449f24f6959b959e49c69aff91a8a14/src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarEntry.cs#L126-L131

Reply:
Ahh, you're correct. I completely forgot we added that throw recently. Nevermind then.